packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -242,6 +280,7 @@ class GrpcRouteSpec extends RouteSpec {
         match: {
           serviceName: this.match.serviceName,
         },
+        timeout: this.timeout ? renderTimeout(this.timeout): undefined,
```

Comment:
You're repeating this expression: `this.timeout ? renderTimeout(this.timeout): undefined` in multiple places. Just make `renderTimeout` handle that case, and the code will just become `timeout: renderTimeout(this.timeout)`.

Reply:
The method signature for `renderTimeout` is currently `CfnRoute.HttpTimeoutProperty`. It makes this method stronger since it will always return this (rather structure type as you've referred to earlier). If you insist I'm glad to do it but im not sure how we get around typescript yelling at us because we cannot return undefined.

Reply:
Simple: change `renderTimeout` to `renderTimeout(timeout?: HttpTimeout): CfnRoute.HttpTimeoutProperty | undefined` 🙂